src/main/java/com/google/api/codegen/transformer/py/PythonImportSectionTransformer.java

```diff
@@ -0,0 +1,173 @@
+  @Override
+  public ImportSectionView generateImportSection(
+      MethodTransformerContext context, Iterable<InitCodeNode> specItemNodes) {
+    return ImportSectionView.newBuilder()
+        .appImports(generateInitCodeAppImports(context, specItemNodes))
+        .build();
+  }
+
+  private List<ImportFileView> generateInitCodeAppImports(
+      MethodTransformerContext context, Iterable<InitCodeNode> specItemNodes) {
+    return ImmutableList.<ImportFileView>builder()
+        .add(generateApiImport(context))
+        .addAll(generateProtoImports(context, specItemNodes))
+        .addAll(generatePageStreamingImports(context))
+        .build();
+  }
+
+  private ImportFileView generateApiImport(MethodTransformerContext context) {
+    return createImport(
+        context.getApiConfig().getPackageName(),
+        context.getNamer().getApiWrapperVariableName(context.getInterface()));
+  }
+
+  private List<ImportFileView> generateProtoImports(
+      MethodTransformerContext context, Iterable<InitCodeNode> specItemNodes) {
+    ModelTypeTable typeTable = context.getTypeTable();
+    Set<ImportFileView> protoImports = new TreeSet<>(importFileViewComparator());
+    for (InitCodeNode item : specItemNodes) {
+      TypeRef type = item.getType();
+      // Exclude map entry types
+      if (!type.isRepeated() && type.isMessage()) {
+        String fullName = typeTable.getFullNameFor(type);
+        String nickname = typeTable.getNicknameFor(type);
+        protoImports.add(generateAppImport(fullName, nickname));
+      } else if (type.isEnum()) {
+        protoImports.add(createImport(context.getApiConfig().getPackageName(), "enums"));
+      }
+    }
+    return new ArrayList<>(protoImports);
+  }
+
+  private List<ImportFileView> generatePageStreamingImports(MethodTransformerContext context) {
+    ImmutableList.Builder<ImportFileView> pageStreamingImports = ImmutableList.builder();
+    if (context.getMethodConfig().isPageStreaming()) {
+      ImportTypeView callOptionsImport =
+          ImportTypeView.newBuilder().fullName("CallOptions").nickname("").build();
+      ImportTypeView initialPageImport =
+          ImportTypeView.newBuilder().fullName("INITIAL_PAGE").nickname("").build();
+      ImportFileView fileImport =
+          ImportFileView.newBuilder()
+              .moduleName("google.gax")
+              .types(ImmutableList.of(callOptionsImport, initialPageImport))
+              .build();
+      pageStreamingImports.add(fileImport);
+    }
+    return pageStreamingImports.build();
+  }
+
+  /**
+   * Orders the imports by: (1) number of attributes (least to most), (2) module name (A-Z), (3)
+   * attribute name (A-Z)
+   */
+  private Comparator<ImportFileView> importFileViewComparator() {
+    return new Comparator<ImportFileView>() {
+      @Override
+      public int compare(ImportFileView o1, ImportFileView o2) {
+        if (o1.types().size() != o2.types().size()) {
```

Comment:
This is one of the relatively few places where I have seen different Python programmers do it different ways. Some people actually put `from <m> import <a>` _above_ `import <m>` to align with the requirement that `from __future__ import anything` be above any other import. Others place that by itself and do what @eoogbe has.

For what it is worth, I think what she has is fine. There is no hard and fast rule here and I have seen a couple different ways to go about it. Consistency is what matters.

For my part, I will probably try to make sure in the stuff that we feed into toolkit that any given module only gets one of the two styles of imports, thus making this sort not matter.